soccer/gameplay/plays/testing/test_adaptive_formation_weights.py

```diff
@@ -0,0 +1,126 @@
+import main
+import play
+import behavior
+import evaluation.field
+import evaluation.passing_positioning
+import evaluation.shooting
+import evaluation.defensive_positioning
+import constants
+import robocup
+import math
+import enum
+
+
+class TestAdaptiveFormationWeights(play.Play):
```

Comment:
We could even do something like parallelizing the execution if we did that (instead of just looping over the entire field, spawn 10 threads, and have each one of them scan over a section, and then return the result), and that would make such visualizations much faster! I think we might even be able to implement something like this on the python side.